b2/raw_api.py

```diff
@@ -457,6 +490,13 @@ def upload_file(
         :param data_stream: A file like object from which the contents of the file can be read.
         :return:
         """
+        # Raise an appropriate exception if the file_name is unusable.
+        try:
+            self.check_b2_filename(file_name)
+        except UnusableFileName as e:
+            sys.stderr.write("\nUpload fails.  Exception is:\n{}\n".format(repr(e)))
+            sys.stderr.write("Exiting.\n")
+            os._exit(1)
```

Comment:
This is where I think we should probably just report and skip the offending file instead of stopping dead.  Let me know if that would be better and I'll change it.  (I left it this way for ease of development, plus I don't yet know how to propagate a non-fatal error up to the overall sync status.)

Reply:
It should just raise the exception.  The ConsoleTool class should catch it and report the error.